src/familybot/lib/steam_itad_mapping_repository.py

```
@@ -0,0 +1,114 @@
+    """
+    if not appids:
+        return {}
+
+    conn = None
+    cursor = None
+    try:
+        conn = get_db_connection()
+        cursor = conn.cursor()
+        placeholders = ",".join("?" * len(appids))
+        cursor.execute(
+            f"SELECT appid, itad_id FROM steam_itad_mapping WHERE appid IN ({placeholders})",
```

Ruff (0.15.7)

[error] Possible SQL injection vector through string-based query construction

(S608)